src/Symfony/Bridge/Monolog/Tests/Handler/FingersCrossed/HttpCodeActivationStrategyDecoratorTest.php

```diff
@@ -0,0 +1,78 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Tests\Handler\FingersCrossed;
+
+use Monolog\Handler\FingersCrossed\ErrorLevelActivationStrategy;
+use Monolog\Logger;
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\Monolog\Handler\FingersCrossed\HttpCodeActivationStrategyDecorator;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class HttpCodeActivationStrategyDecoratorTest extends TestCase
+{
+    public function testExclusionsWithoutCode()
+    {
+        $this->expectException('LogicException');
+        new HttpCodeActivationStrategyDecorator(new ErrorLevelActivationStrategy(Logger::WARNING), new RequestStack(), [['urls' => []]]);
+    }
+
+    public function testExclusionsWithoutUrls()
+    {
+        $this->expectException('LogicException');
+        new HttpCodeActivationStrategyDecorator(new ErrorLevelActivationStrategy(Logger::WARNING), new RequestStack(), [['code' => 404]]);
+    }
+
+    /**
+     * @dataProvider isActivatedProvider
+     */
+    public function testIsActivated($url, $record, $expected)
+    {
+        $requestStack = new RequestStack();
+        $requestStack->push(Request::create($url));
+
+        $strategy = new HttpCodeActivationStrategyDecorator(
+            new ErrorLevelActivationStrategy(Logger::WARNING),
+            $requestStack,
+            [
+                ['code' => 403, 'urls' => []],
+                ['code' => 404, 'urls' => []],
+                ['code' => 405, 'urls' => []],
+                ['code' => 400, 'urls' => ['^/400/a', '^/400/b']],
+            ]
+        );
+
+        $this->assertEquals($expected, $strategy->isHandlerActivated($record));
+    }
+
+    public function isActivatedProvider()
+    {
+        return [
+            ['/test',  ['level' => Logger::ERROR], true],
+            ['/400',   ['level' => Logger::ERROR, 'context' => $this->getContextException(400)], true],
+            ['/400/a', ['level' => Logger::ERROR, 'context' => $this->getContextException(400)], false],
+            ['/400/b', ['level' => Logger::ERROR, 'context' => $this->getContextException(400)], false],
+            ['/400/c', ['level' => Logger::ERROR, 'context' => $this->getContextException(400)], true],
+            ['/401',   ['level' => Logger::ERROR, 'context' => $this->getContextException(401)], true],
+            ['/403',   ['level' => Logger::ERROR, 'context' => $this->getContextException(403)], false],
+            ['/404',   ['level' => Logger::ERROR, 'context' => $this->getContextException(404)], false],
+            ['/405',   ['level' => Logger::ERROR, 'context' => $this->getContextException(405)], false],
+            ['/500',   ['level' => Logger::ERROR, 'context' => $this->getContextException(500)], true],
+        ];
+    }
+
+    protected function getContextException($code)
```

Comment:
This method should be private.

```suggestion
    private function getContextException(int $code): array
```

Reply:
done